test/lib/rules/no-template-children.js

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+
+const { RuleTester } = require('eslint');
+
+const { ESLINT_TEST_CONFIG } = require('../shared');
+const rule = require('../../../lib/rules/no-template-children');
+
+const ruleTester = new RuleTester(ESLINT_TEST_CONFIG);
+
+function makeCode(property, style) {
+    switch (style) {
+        case 'direct':
+            return `const foo = this.template.${property};`;
```

Comment:
`const foo = this['template']` is another way of accessing it.

Reply:
In this case, you are really asking for trouble 😆 .

Reply:
🤣 I know, I was just pointing it out.